Develop Branch for CI/CD

ghost opened the pull request.

garywong-bc commented on migrations/initdb-postgis.sh:
```diff
@@ -0,0 +1,23 @@
+	"${psql[@]}" --dbname="$DB" <<-'EOSQL'
+		CREATE EXTENSION IF NOT EXISTS postgis;
+		CREATE EXTENSION IF NOT EXISTS postgis_topology;
+		CREATE EXTENSION IF NOT EXISTS fuzzystrmatch;
```
FYI our PostGIS [image](https://github.com/bcgov/openshift-postgresql-oracle_fdw/blob/bbf74c4f571c8d064c3c300f5ded026ade04eff2/rhel7.rh-postgresql96/Dockerfile#L121) doesn't have all the PostGIS-related extensions .. if we need them we can add them too, but Clecio will have to re-generate a `new v2_stable`

exzizt committed:
MDS-3132: Fix NoW View Status Column Filter (#1612)

Main
MDS-3132
Fixes an issue where filtering by status in the Notice of Work view did not work correctly
Screenshot
image

MDS-3332
Fixes an issue where it was possible to navigate to an invalid NoW tab by changing the URL
Other
Updates our NoW status badge stylings
Fixes an issue where Pendering Verification was set in the code as a possible vFCBC status

nayr974 committed:
Update V2021.01.12.14.09__remove_additional_conditions_category.sql

nayr974 committed:
Revert "Update V2021.01.12.14.09__remove_additional_conditions_category.sql"

This reverts commit 4a507b0363da76c8e747f0e3dc8c2c225e96ea97.

nayr974 committed:
Create V2021.01.12.14.08__before_remove_additional_conditions_category.sql

nayr974 committed:
Create V2021.01.12.14.07__before_remove_additional_conditions_category.sql

Commit:
Updating develop with master

mhuseinov committed:
MDS-3314: added generate permit number step before issue permit (#1614)

* MDS-3314: added generate permit number step before issue permit

* MDS-3314: tidy

* MDS-3314: common folders

* MDS-3314: pr comments

mmumby committed:
Change integers to numeric for NoW columns (#1613)

JustinMacaulay committed:
fix to make permitee history show all permitees, past and present (#1617)

mmumby committed:
Misc bug fixes (#1618)

exzizt committed:
MDS-3308: Users Can Add Author, Title, and Date Info For Final Application Package Documents & Previous Amendment Documents (If Applicable) (#1619)

Users can edit metadata (title, author, date) for documents in the Final Application Package and for documents related to the previous permit (if applicable) in the Draft Permit tab while processing a permit. Title is mandatory while author and date are not.

Commit:
Updating develop with master

pdittaro committed:
Disable MMS ETLs (#1620)

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

exzizt committed:
Fix save NoW draft bug (#1621)

* More safely combine final app package docs

* Check if isEmpty first

mmumby committed:
MERGE FOR LAUNCH - Remove Feature flagging on NoW and regional mine editing (#1607)

* Removed FE feature flagging for NoW and regional mine editing

* updated snapshot tests

Co-authored-by: Ryan Rondeau <nayr974@users.noreply.github.com>

exzizt committed:
IN 'DATA'! (#1623)

pdittaro committed:
Fix for ETLS in Test (#1624)

* WIP

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Add if condition

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Second half of ETL fix

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

mmumby committed:
MISC BUG FIXES  (#1625)

* added a few FE data checks

* Update NOWReviewForm.js

* updated the permittee address update link to point to the permittee

* removed received date as required on referrals

* Update NOWReviewForm.js

* disable starting of technical review before assigning lead inspector

* Update ProcessPermit.spec.js.snap

* MDS-3314: unit tests

* updated linting

* updated snapshot tests

* added a migration to remove duplicate doc types.

* Fix for "When I click the preview button for a draft permit, the section I was working in collapses in the background."

* updated banner styling to always be on the bottom

* updated table disclaimers

* fix for template footer

* updated snapshot

* updated props

* updated security snapshots

* updated logic of securities on permit amendment modal

* Fix "Contact Confirm button seems to need to be clicked 2x in order to trigger visual confirmation. On first click it seems to briefly show the required field validation text and then return to unclicked state"

* removed required field from security not required UI

* updated

* updated loadingWeapper css, and snapshots

* updated snapshots

* updated linting/props

* updated snapshots

* updating postal code validation to allow american zip codes.

* updated fetch on permit tab to include party relationships

Co-authored-by: Ryan Rondeau <ryan_rondeau@live.ca>
Co-authored-by: mhuseinov <mekhti.huseinov.ssi@gmail.com>
Co-authored-by: justinmacaulay <justinkmacaulay@gmail.com>
Co-authored-by: mhuseinov <61513701+mhuseinov@users.noreply.github.com>
Co-authored-by: Luke Mitchell <lukegamitchell@gmail.com>

nayr974 committed:
Update mms_now_import.py

mhuseinov committed:
MDS-3311: fix for issue date on draft permit (#1626)

mmumby committed:
Added fetch mine to permit tab (#1627)

* added mine fetch to permit tab on initial mount

* added mine fetch

* updated snapshots

* updated snapshots

* removed permit method duplicate

Co-authored-by: Ryan Rondeau <nayr974@users.noreply.github.com>

nayr974 committed:
Update R__view_notice_of_work.sql

mhuseinov committed:
MDS-set processed field tto 'Y' (#1628)

Co-authored-by: Ryan Rondeau <nayr974@users.noreply.github.com>

Commit:
Updating develop with master

exzizt committed:
MDS-3110: Users can view a mine's MMS files (using Syncfusion File Manager) (#1602)

* POF for documents MMS backdoor

* updated

* updated linting, WIP feature

* updated snapshots

* Add syncfusion dependencies to project

* Remove unneeded code

* Bring in sample code

* Remove unneeded code

* Prep core-web env with object store vars

* Import their css file

* Import their css

* Setup and use their demo s3 file bucket

* Whitespace

* We are going to have C# in our project perhaps. WOOO!

* Update dockerfiles/configs

* Formatting

* Create openssl.cnf

* Update .gitignore

* Formatting

* Get it working

* Tidy

* Create .env-example

* Add default prop params

* Remove file modification actions from default props

* Change text

* Delete upload method

* Delete file/folder modification actions

* Tidy

* Reduce syncfusion packages to minimum required, tidy

* Update package jsons

* Load syncfusion scss correctly!

* Grab theme override variables

* Do some theme overrides

* Update comment URL

* Use DotNetEnv package and load env variables correctly

* Change where menu lives

* Setup env vars for front end, make sure that root is always the mine folder ONLY, other

* Update DashboardRoutes.spec.js.snap

* Remove comment

* Tidy file

* Update .gitignore

* Delete CHANGELOG.md

* Delete TestController.cs

* Delete EJ2AmazonS3ASPCoreFileProvider.sln

* Install DotNetEnv more correctly

* Set up Syncfusion license registration

* Tidy

* Update syncfusion packages

* Provide auth for file operations endpoint

* Set up Keycloak authorization

* Authorize download endpoint using workaround, add more default props

* Remove newline

* Trigger download for file open except for images and folders

* add templates for filesystem-provider

* Update filesystem-provider.bc.json

* Update filesystem-provider.bc.json

* add proxy for file api

* Update filesystem-provider.bc.json

* Update filesystem-provider.bc.json

* Update filesystem-provider.bc.json

* Update filesystem-provider.bc.json

* Update webpack.parts.js

* Change how JWT OIDC Authority and Audience env vars are being read

* Update filesystem-provider.bc.json

* update versions

* Update EJ2AmazonS3ASPCoreFileProvider.csproj

* Update EJ2AmazonS3ASPCoreFileProvider.csproj

* Update EJ2AmazonS3ASPCoreFileProvider.csproj

* Update EJ2AmazonS3ASPCoreFileProvider.csproj

* Update EJ2AmazonS3ASPCoreFileProvider.csproj

* version

* Update config-build.groovy

* Image stream in DC namespace

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Remove Route declaration

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Escape variable in groovy script

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Environment variables

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Update R__view_notice_of_work.sql

* Fix variable navigation

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Reduce compute request

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Env variables and remove explicit syncfusion reference

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Use proxy to truncate?

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Remove double template parameter entry

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Accidentally reverted API_URL

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Add tail slash

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Include file-api nginx block

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Deploy to Test and Prod configuration

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Include protocol in nginx proxy request

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Prefix module name for variable sub

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Remove smurf naming on nginx upstream

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Remove suffix of upstream nginx

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* JWT Audience for FS

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Include JWT Audience in test and prd

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Change to /file-api location on filesystem provider

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* File-api location

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Disable HTTPS redirect

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Disable request buffering

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Renable HTTPS between proxy and fs

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Reenable buffering

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Enable Swagger

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Remove Swagger

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Remove swashbuckle dependencies

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Add back swagger

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Enable endpoint validation

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Update Post

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Increase logging to Debug

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Comment out code related to Authorization

* Disable HSTS

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Make it all null

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Update AmazonS3ProviderController.cs

* Raw string

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Alternate way to read POST body method test

* File-api wtf

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Add fs env variable to frontend test prod

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Common folders

Co-authored-by: Meredith Mumby <meredith@freshworks.io>
Co-authored-by: Ryan Rondeau <ryan_rondeau@live.ca>
Co-authored-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Include FS API URL in CD nginx config

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Consistent module naming across envs

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Add JWT Authority validation

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Commas 0_o;;

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

Commit:
Updating develop with master

exzizt committed:
Update syncfusion.scss (#1629)

Remove "Drag and drop to upload" message for empty folders

mmumby committed:
updated staticContent fetch (#1630)

mmumby committed:
updated varchar lengths of post_code to match the new validations, and updated suite_no to a max of 10 to meet user needs (#1631)

nayr974 committed:
increase length of currency validation

Commit:
Updating develop with master

nayr974 committed:
Update ReferralConsultationPackage.js

nayr974 committed:
Update now_application_resource.py

Commit:
Updating develop with master

nayr974 committed:
Update permit.py

Commit:
Updating develop with master

mmumby committed:
Bug/change new permit to amendment (#1632)

nayr974 committed:
Update R__view_notice_of_work.sql

mmumby committed:
Bug/mds 3369 (#1633)

nayr974 committed:
Update models_mixins.py

nayr974 committed:
MDS 3360 (#1635)

* Update BondForm.js

* Update BondForm.spec.js.snap

Commit:
Updating develop with master

mmumby committed:
Changed create draft permit component (#1636)

Commit:
Updating develop with master

nayr974 committed:
Update BondForm.js

nayr974 committed:
Update BondForm.spec.js.snap

Commit:
Updating develop with master

exzizt committed:
MDS-3354: Fix Name Issue With Party Organizations (#1637)

Main
Fixes an issue where parties of type "organization" could be created with a "first name" field, causing data quality issues
Sets first_name to null for all existing parties that are an organization
Other
Fixes an issue where a party's "name" hybrid property and expression was not being correctly calculated

nayr974 committed:
Update nris_data_import.py

mhuseinov committed:
MDS-increase max_connections value (#1638)

Co-authored-by: Ryan Rondeau <nayr974@users.noreply.github.com>

Commit:
Updating develop with master

Jsyro committed:
Issue permits as vc with mdt (#1634)

* everything except config for api call that issued VC

skips if permittee is not related to orgbook entity

* file permission needs to be executable

* config, placeholder apikey

* first pass happy path with FE.

* snapshot

* renaming config

* folder sync

* no button, and run liveness check first.

Could build the service out into a proper singleton initialized on app start up with a threaded init/readyness check

* print on 'not live' retry

* better logging

* two tests.... but module drilling and attribute access got confused.

* rename service and added attempted VC in permit issuance

Signed-off-by: Jason Sy <jasyrotuck@gmail.com>

* missed param

* comment

* error if explicit call does not complete properly.

* remove unused imports

* needed this one.

* add context to popconfirm

* get actual inspector name.

* release this to admins only.

* throw action creator errors

* empty response bad.

* remove verb from references to resource

* hard code values for major mines, try your best on regional.

* always use set name if exists.

* formatting idfference

* permitee name and xref relationsihps

* super obtuse error.

* attributes

* typ

* rename

Co-authored-by: Ryan Rondeau <nayr974@users.noreply.github.com>

mhuseinov committed:
MDS-3318: fix for issue VC (#1640)

exzizt committed:
MDS-3394: Fix file uploads sometimes failing due to resumable upload feature (#1639)

Main
Fixes an issue where file uploads would sometimes fail if they were being resumed and their stored cached data had expired and/or the object store responded with "not found" and we didn't properly return that status
Implements in our TUS API the "upload expires" header to indicate to the client when our cached upload metadata expires
Shows an error notification for file upload errors:
image
Other
Adds a new accepted spatial filetype

mmumby committed:
Added units to "estimate_rate" & updated CoreEditableTable (#1641)

mmumby committed:
added a default to unit if null, removed undefined from UI (#1642)

exzizt committed:
MDS-3405: Add more spatial file types & allow images to be uploaded to a NoW (#1643)

Main
Adds more spatial file types
Allows images to be uploaded to a NoW
Other
Updates or .gitignore to ignore some files that VSCode generated for our C# files

Commit:
Updating develop with master

mmumby committed:
Update field size NoW (#1644)

pdittaro committed:
Digdag fix

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Merge remote-tracking branch 'origin/develop' into develop

pdittaro committed:
Image namespace fix

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

Commit:
Updating develop with master

pdittaro committed:
Fix non-executing ternary assignment

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

mmumby committed:
Fix - added camps to reclamation summary (#1646)

pdittaro committed:
Digdag fix

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

jon-funk committed:
Double unittest timeout in sonar pod (#1652)

exzizt committed:
MDS-3373/MDS-3419: Fix NoW small number values not saving correctly & Fix broken report filtering by "received only" (#1649)

Main
MDS-3373
Removes the precision/scale restrictions for NoW disturbed_area and total_disturbed_area fields
MDS-3419
Fixes broken filtering by "received only" for reports
Other
Allowed number of decimals for estimate_rate was previously increased but this was not reflected in all areas
Changes m3 --> m³ in some places and minor other text changes

mhuseinov committed:
Feature/mds 3402 flower authentication (#1651)

* MDS-3402:authentication to flower

* MDS-3402: updated env-example

* MDS-3402: flower url configuration

* MDS-3402: missing parameter

* typo

* typo

* MDS-3402: updated groovy files

* MDS-3402: typo

mhuseinov committed:
Mds 3318  contacts   tenure holder, land owner, site operator management (#1645)

* MDS-3318: fix for issue VC

* MDS-3318: backend WIP, fix of white screen of death on NOW

* MDS-3318: display permit nuber on contact card on mine page

* MDS-3318: adding contacts on Mine page

* MDS-3318: added permit to test

* MDS-3318: label for not assigned permetie, fix merge

* MDS-3318: permit id check

mhuseinov committed:
Feature/mds 3403 monitoring for flower workers (#1654)

* MDS-3403: livenessProbe for workers

* Specify correct celery app

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

Co-authored-by: Paul Dittaro <pdittaro@gmail.com>

jon-funk committed:
MDS-3347: Get a data dump of test locally

jon-funk committed:
Merge branch 'develop' into MDS-3347-Get-a-data-dump-of-test-locally

mhuseinov committed:
MDS-3438: updated docs footer (#1653)

jon-funk committed:
Add dump cleanup and gzip to reduce footprint

jon-funk committed:
Refactor to atomic commands and updated usage

jon-funk committed:
Merge branch 'develop' into MDS-3347-Get-a-data-dump-of-test-locally

jon-funk committed:
Merge pull request #1656 from bcgov/MDS-3347-Get-a-data-dump-of-test-locally

MDS-3347: Get a data dump of test locally

mhuseinov committed:
Feature/mds 3456 (#1658)

* MDS-3456: update contact supports permit selection for TH, SO, LO

* MDS-3456: update date range validation logic

* MDS-3456: improvements to party assignment

exzizt committed:
MDS-3426: "Fix permit draft incorrectly attached as the issued permit" (#1657)

Main
Fixes an issue where, when issuing a permit, the generated permit PDF was not always attached as a related document to the permit as intended
Other
Fixes an issue where the "submitting" indicator was not correctly being set in UpdateStatusGenerateLetterModal.js and the user could spam click the button
Sets the disabled prop for some buttons when submitting
Minor adjustments to comments and documentation

nayr974 committed:
include contact if missing phone number (#1655)

pdittaro committed:
Use different credentials for NROS document download

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Merge remote-tracking branch 'origin/develop' into develop

mhuseinov committed:
Feature/mds 3374 unit type in core (#1662)

* Create V2021.02.18.11.44__rename_document_type.sql

* Change document name from Technical Review --> Notice of Work Form

* MDS-3374: added water supply units to Edited application pdf

* MDS-3374: access road snapshot

* MDS-3374: snapshot test

* MDS-3374: fix for editable value unit column

* MDS-3374: updates for editable table, updated undergraound exploration default unit_type_code

* MDS-3374: updated Placer Operations

* MDS-3374: make Estimate Rate required field on Water supply

* MDS-3374: added space to titles, upadated tests

* MDS-3374: updated cut lines title

jon-funk committed:
Fix/increase-flyway-deploy-mem

mhuseinov committed:
fix for flower auth (#1665)

Co-authored-by: Paul Dittaro <pdittaro@gmail.com>

jon-funk committed:
Merge branch 'develop' into Fix/increase-flyway-deploy-mem

jon-funk committed:
Merge pull request #1666 from bcgov/Fix/increase-flyway-deploy-mem

Fix/increase-flyway-deploy-mem

exzizt committed:
MDS-3428: Add Notice of Work Form Automatically to FAP (#1661)

Main
When a Notice of Work is imported or every time Technical Review is completed, a Notice of Work Form document is automatically generated and included in the final application package. All previous Notice of Work Form documents that were in the final application package are removed from it.
Renames the document Technical Review to Notice of Work Form.
Notice of Work Form documents are no longer excluded from the NoW documents lists.
image
Other
Sets a "submitting" state on the "Update NoW Progress" modal so the buttons cannot be spammed while submitting.

mmumby committed:
Verify NoW Contacts and Update CORE contacts (#1660)

mmumby committed:
misc styling tweaks (#1667)

mhuseinov committed:
Feature/mds 3374 unit type in core (#1663)

* Create V2021.02.18.11.44__rename_document_type.sql

* Change document name from Technical Review --> Notice of Work Form

* MDS-3374: added water supply units to Edited application pdf

* MDS-3374: access road snapshot

* MDS-3374: snapshot test

* MDS-3374: fix for editable value unit column

* MDS-3374: updates for editable table, updated undergraound exploration default unit_type_code

* MDS-3374: updated Placer Operations

* MDS-3374: make Estimate Rate required field on Water supply

* MDS-3374: added space to titles, upadated tests

* MDS-3374: updated cut lines title

* MDS-3374: set default values for length, width, height, migration for historical data

* MDS-3374: updated migration file name

mhuseinov committed:
Bugfix/mds 3415 (#1668)

* MDS-3415: removed length limitations on party table

* MDS-3415: added extension validation on UI

* MDS-3415: reset length liitation on address table

* MDS-3415: updated snapshot test

* MDS-3415: added length validation to contacts first and last name on front-end

* MDS-3415: tidy

* MDS-3415: updated common folder

mhuseinov committed:
Bugfix/mds 3495 (#1670)

* MDS-3415: removed length limitations on party table

* MDS-3415: added extension validation on UI

* MDS-3415: reset length liitation on address table

* MDS-3415: updated snapshot test

* MDS-3415: added length validation to contacts first and last name on front-end

* MDS-3415: tidy

* MDS-3415: updated common folder

* MDS-3495: moved phone number validation on resource level, added phone validation on Verify NOW

* MDS-3495: updated error message

* MDS-3495: updated error message

* MDS-3495: updated error message

* added scroll class to contact rows

* updated size of information cards

* updated the content card to fit in the scroll

* MDS-3495: removed redundant BE tests

Co-authored-by: Meredith Mumby <meredith@freshworks.io>

pdittaro committed:
Fix: Flower host to pipeline configuration

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

Commit:
Updating develop with master

mmumby committed:
FIx- Add NROS/vFCBC tooltip to "historic" applications (#1650)

nayr974 committed:
MDS-3173 - GIS Spatial Files + MDS-3466 - Editing FAP Throws Error  (#1669)

* api work

* add roles

* Update NOWPermitGeneration.js

* Update test_expected_auth.py

* Update test_expected_auth.py

* Create V2021.02.27.12.30__update_now_view.sql

* add safety to backend permit amendment document metadata

Co-authored-by: Meredith Mumby <meredith.mumby@gmail.com>

pdittaro committed:
MDS-3396 Permit documents should show the total security amount not the adjusted security amount (#1664)

jon-funk committed:
bugfix/increase-flyway-dbbackup-memory (#1673)

exzizt committed:
Fix can't save empty string preamble date (#1674)

Fixes an issue where sending over an empty preamble date was sending over an empty string, and the backend could not parse this correctly as null

Commit:
Updating develop with master

nayr974 committed:
Add "DRAFT" to filename based on is_draft template data; Fix forEach loop; (#1672)

* add draft if draft document
* Set loading/disable prop for buttons in Start Draft Permit modal

exzizt committed:
MDS-3487: Add More CRR Document Types (#1675)

Main
Adds some more CRR documents types

Commit:
Updating develop with master

jon-funk committed:
Fix: update docgen api image ver to 2.1.2 to increase max request size (#1676)

exzizt committed:
Increase local docgen to 2.1.2 (#1679)

Commit:
Updating develop with master

mmumby committed:
Update: Placer data/Reclamation Summary Calculations, misc pdf updates/formatting (#1671)

exzizt committed:
Enhancement/Modify word-wrap for toasts to show whole messages & String replace psycopg2 (#1680)

Modifies the style of our notification messages to better fit long text
Replaces occurrences of "psycopg2" with "Database Error" for our "custom axios" error messages

mhuseinov committed:
Bugfix/mds 3420 (#1677)

* MDS-3420: fix for status progress endpoint

* MDS-3420: migration to update wrong statuses

* MDS-3420: split data clean up

mhuseinov committed:
Bugfix/mds 3420 (#1682)

* MDS-3420: part2: migration to update historical data

mmumby committed:
Removed water_supply reclamation description (#1681)

mmumby committed:
Update/now pdf (#1683)

mmumby committed:
Update/placer units (#1685)

mhuseinov committed:
update now view to map more statuses (#1684)

nayr974 committed:
GIS NoW Export (#1678)

* api work

* add roles

* Update NOWPermitGeneration.js

* Update test_expected_auth.py

* Update test_expected_auth.py

* Create V2021.02.27.12.30__update_now_view.sql

* add safety to backend permit amendment document metadata

* add draft if draft document

* Set loading/disable prop for buttons in Start Draft Permit modal

* Create R__view_now_gis.sql

* WIP GIS Export

* Use csv module

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Formatting

* Add GIS export endpoint to namespace

* Fix incorrect column reference in join

* Add in NoW progress to GIS export view

* Rearrange column, add NoW long/lat

* Add adjusted/proposed tonnage

* Import update and formatting of csv endpoint

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Add term_of_application

* Update model

* Add todo comment

* Update now_application_gis_export_resource.py

* Update resource name/inclusion

* Fix incorrect property names

* Undo set cache to null

* Shameless copy of GIS resource for GIS file resource

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Remove file_export csv endpoints

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

Co-authored-by: Luke Mitchell <lukegamitchell@gmail.com>
Co-authored-by: Luke Mitchell <exzizt@users.noreply.github.com>
Co-authored-by: Meredith Mumby <meredith.mumby@gmail.com>
Co-authored-by: Paul Dittaro <pdittaro@gmail.com>

mmumby committed:
Update/pdf to include more water supply details (#1686)

Commit:
Updating develop with master

mmumby committed:
updated the styling on map footer to show the mine location (#1687)

nayr974 committed:
Minor changes to GIS Export (#1688)

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Update now_application_gis_export_resource.py
* Updates to GIS export
* Update now_application_gis_export_resource.py

nayr974 committed:
Fix group by issue in export query

exzizt committed:
MDS-3437: Standardize Document Template Fonts (#1690)

Main
Standardizes all of our document template fonts to be Arial
image
Note
Completion of this ticket required making a PR into the Common Services team's repo and them creating a new docgen-api version, 2.1.3.

Commit:
Updating develop with master

mmumby committed:
Log vfcbc data in test (#1693)

mmumby committed:
Feature/mds 3136 (#1692)

pdittaro committed:
MDS-3557 Update Matomo URLs (#1691)

* Gitignore update

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Update Matomo Urls to OCP4

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

mhuseinov committed:
MDS-3136: updates (#1694)

MDS-3136: administrative amendments updates

pdittaro committed:
Email on failure of NRIS etl job (#1695)

* Test email on failure of NRIS etl job

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Add additional required mail parameters

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Remove emails

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

exzizt committed:
MDS-3541: Create Admin-Only Button to Trigger NoW Document Import Job (#1696)

Main
Creates an admin-only button to trigger the import-job for a NoW's submission documents
image
Other
Fixes an issue where we were incorrectly serializing success_docs of task_result in the import NoW submission documents job, causing the job to remain "in progress" the first time it ran even if it was successful

exzizt committed:
Fix/minespace spatial file uploads (#1698)

* WIP code to trigger import now doc job

* Fix task_result serialization error

* Add reimport button

* Update common folders

* Update required role to admin

* Update toast text

* Remove debug, update Specs

* Remove debug

* Add 1 sec delay to get better job status

* Remove TODO

* Update common

* Update filepond

* Update FileUpload component

* Update file types

* Update allowed file types for report uploads

nayr974 committed:
Update index.html

nayr974 committed:
Update index.html

nayr974 committed:
Update index.html

mhuseinov committed:
Feature/mds 2137 collect government agency info (#1697)

* MDS-2137: government agency info on mine

nayr974 committed:
Update index.html

nayr974 committed:
MDS-3575 - Add letter body content to document templates. (#1699)

* Add letter body content to document templates.

* fix json in script update, update seed data

Co-authored-by: Luke Mitchell <exzizt@users.noreply.github.com>

nayr974 committed:
add missing quote

pdittaro committed:
Hardcode psql repo

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Add rpm suffix

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Install specific rpm file needed

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Update to latest openjdk image

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Updating all tag references to match new flyway tag

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Revert to original

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Flyway build fix (#1702)

Use UBI rather than RHEL

pdittaro committed:
Change reference of antd component (#1703)

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Overload app.sh to run ETL mode

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

pdittaro committed:
Merge branch 'fix/nris-job' into develop

Commit:
Updating develop with master

exzizt committed:
Update afterMigrate__1__seed_data.sql (#1707)

exzizt committed:
MDS-3589: Add NoW Number to Generated NoW Document Names (#1704)

Main
Adds the NoW Number to the beginning of the document name for all generated NoW documents
image

exzizt committed:
MDS-3608: Update NoW Document Types (#1700)

Main
Sets a sub-type for various NoW document types
Adds a couple of new NoW document types (Application Export Files and Additional Application Files)
Adjusts where certain sub types of NoW documents can be displayed/uploaded
image

image

Other
Fixes some typos in the description of some NoW document types

mmumby committed:
Enable internal communication on mine record (#1708)

pdittaro committed:
Hardcode volume mount for NRIS ETL

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

mhuseinov committed:
MDS-added filter based on now type (#1706)

* MDS-added filter based on now type

mmumby committed:
Move Site Properties to permit/Add Exemption Fee status (#1705)

jon-funk committed:
Revert "Move Site Properties to permit/Add Exemption Fee status (#1705)" (#1710)

This reverts commit cd7da4273af59ca3c786edca0ab7a4751cf1ef98.

mhuseinov committed:
Feature/mds 3338 add status for withdrawn (#1709)

add status for withdrawn for a NOW

exzizt committed:
MDS-3566: Updates to Notice of Work Form Document (#1711)

Main
Removes the "exported" and "lasted updated by" fields from the NoW Form document
image
Fields and sections in the NoW Form document are now organized in a similar order to the sections in the vFCBC application form
Some sections in the NoW Form document have been renamed to match the sections in the vFCBC application form
Many sections in the NoW Form document are now only rendered if they contain at least one filled-in field and that section is applicable to the NoW type
Sections in the "Reclamation Summary" table are now only populated if data exists for that section
Fixes an issue where many fields in the NoW Form were not being populated (e.g., the entire "camps" section)
image
Other
Adds "temporary Microsoft files" to our gitignore
Notes
Sometimes fields display "EDITED" even though they don't appear to be is due to strange whitespace received in some of the data from vFCBC:
image

mhuseinov committed:
Feature/mds 3572 (#1712)

MDS-3572: site properties

mmumby committed:
Bug/document upload (#1716)

Commit:
Updating develop with master

exzizt committed:
sand_and_gravel --> sand_gravel_quarry_operation (#1718)

Main
Fixes an issue where the "Sand & Gravel / Quarry Operations" section was not being rendered in the NoW form document, due to inconsistent naming of the NoW activity (the activity name is "sand_gravel_quarry_operation" but we were sometimes referencing it as "sand_and_gravel", which is not the actual activity name.
image
Other
Removes the "Permit Application Fee Assessment" section from the NoW form document and adds in a "Maximum Annual Tonnage" section.
image

mmumby committed:
Bug/comment permissions (#1719)

mmumby committed:
Bug/parties permissions (#1720)

Commit:
Updating develop with master

mmumby committed:
Hackathon/tailings (#1717)

exzizt committed:
MDS-3571: Updates to "Mine Summary View" Endpoint (Requested by GenTax) (#1722)

Main
Adds to the mine summary view:
Permit exemption fee
Permit site properties
Permit ID
Permit status code
Permit status changed timestamp
Mine government agency type fields
Adds a "status changed" timestamp column to the permit table to capture when the permit's status changes
Other
Fixes an issue where multiple rows could be returned per permit for the mine summary view, due to improperly selecting the permittee and also by not properly grouping by the permit ID in the GROUP BY clause

pdittaro committed:
Hotfix for mine_disturbance_codes safe access

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

mhuseinov committed:
Feature/mds 3629 site properties for now and ada (#1721)

site properties for NOW and ADA, fix for a close permit, fix for commodities on mine list

mmumby committed:
Bug/transfer bond (#1724)

mmumby committed:
Feature/mds 3607 (#1723)

Commit:
Updating develop with master

mhuseinov committed:
MDS-fix for rejection and withdrawal, removed unecessary validation for generate draft permit form (#1727)

Commit:
Updating develop with master

mhuseinov committed:
Feature/mds 3610 allow users to change between now types (#1728)

allow users to change between now types

mmumby committed:
updated the permit add and edit forms to not require auth end date if a Major Mine permit, ie Coal or Mineral. (#1730)

mhuseinov committed:
Feature/mds 3610 allow users to change between now types (#1732)

exzizt committed:
MDS-3706: Swaggerize the Mine Summary Resource (#1733)

Main
Updates the "mine summary" resource so that it provides API documentation
image

JustinMacaulay committed:
MDS-3297 no permit required (#1731)

exzizt committed:
MDS-3693: Fixes to NoW "Total Merchantable Timber Volume" (#1735)

Main
MDS-3693
Fixes an issue where the "Total Merchantable Timber Volume" was an editable field/value in Core instead of being a derived value
image
MDS-3720
Fixes an issue where newly added NoW activities with reclamation data weren't being displayed in the Reclemation Summary Table
image
image
Other
Removes duplicate property from NoW application model

mmumby committed:
Administrative Amendments Rework (#1734)

Commit:
Updating develop with master

nayr974 committed:
Update dependencies to address critical vulnerabilities (#1729)

* update dependencies

mmumby committed:
Bug/permission updates delete permit amendment (#1736)

Commit:
Updating develop with master

nayr974 committed:
Add file type option to generated NoW documents. (#1738)

* add docx option to form templates

* Update GenerateDocumentForm.js

mmumby committed:
added an additional check to the permit table to filter out draft permits when displaying the last amended date (#1737)

nayr974 committed:
Permittee relationship incorrect on Permit model (#1739)

* add docx option to form templates

* Update GenerateDocumentForm.js

* contact changes

* Revert "contact changes"

This reverts commit 07eefdb1b4e997e9ef503187b1dcafbdc15cdaef.

* add permit appointments to permit model

Co-authored-by: Luke Mitchell <exzizt@users.noreply.github.com>

exzizt committed:
MDS-3652: PDF Viewer Implementation (#1725)

Main
Implements a Document Viewer component to display documents within the application.
Currently, this is implemented for PDFs and uses Syncfusion's PDFViewer component. In the future, we can use Syncfusion's other components for displaying "word" and "sheet" documents in the app as well.
All documents that are openable by the Document Viewer are opened instead of being downloaded. This is the case for all documents that exist within Core, and not for NoW submission documents that have not yet been imported, for example.
Screenshots
image

Other
Creates and implements a Document Link component for consistently displaying document links.
Tidying of various code (e.g., reduce code duplication in actionlessNetworkCalls.js.

mmumby committed:
Updated/vfcbc data (#1740)

mhuseinov committed:
Feature/mds 3749 (#1742)

* MDS-3749: NOW equipment section

mhuseinov committed:
MDS-3665: replaced exploration checkbox to radio buttons (#1744)

mhuseinov committed:
MDS-3755 added permit enclosed letter to generate document list (#1745)

mhuseinov committed:
MDS-3749: added serial to equipment in now_submissions (#1746)

* MDS-3749: added serial to equipment in now_submissions

* MDS- updated migration with schema parameter

* updated ownership for seq

Commit:
Updating develop with master

mhuseinov committed:
MDS - update reclamation cost precision (#1747)

mhuseinov committed:
Feature/mds activity summary precision update (#1749)

* MDS: updated response model

mhuseinov committed:
Bugfix/mds 3611 (#1750)

* MDS-3611: exclude signature from put for non-admin user

exzizt committed:
MDS-3768: Document Link Upgrade (#1751)

Main
Users can choose to download a document directly or open it in the Document Viewer (if possible)
image

Commit:
Updating develop with master

mmumby committed:
Bug/withdrawal status (#1752)

mmumby committed:
Update/vfcbc submission data (#1743)

JustinMacaulay committed:
MDS-3722 Docgen and Document Manager running locally (#1741)

mmumby committed:
Bug/vfcbc data types (#1754)

exzizt committed:
MDS-3646: Users Can Create Work Information (Start/Stop) Records in Core (#1753)

Main
Users can now create/edit/delete "work information" records in Core. These records will contain the work "start/stop" data as well as other data in the future.
image
image

image

Users can see and filter by the mine work status on the mine view.
image
Other
Changes the order of the columns in the Browse Mines view

JustinMacaulay committed:
MDS 3584 preview docs before generating (#1756)

mmumby committed:
updated the transmogrify function to handle numberofsites when null from MMS (#1758)

Commit:
Updating develop with master

JustinMacaulay committed:
BUG: Missing function values for the preview document. (#1759)

Commit:
Updating develop with master

exzizt committed:
MDS-3670: GIS Export Updates (#1757)

Main
Adds some new columns to the GIS Export View:
Mine work start date
Mine work stop date
Mine last inspected date
Mine last inspection type
NoW total disturbed area
NoW total disturbed areas for each activity type
Number of days the NoW is in Client Delay
Other
Also adds messageid and mms_cid
Fixes the naming of some "afterMigrate" migrations so that they run in the intended order (e.g., 100 was running before 1)

mmumby committed:
Upgrade/camps activity (#1760)

mmumby committed:
updated the transmografy to handle a boolean value (#1762)

mmumby committed:
Added 'other_information' to NoW UI (#1764)

jon-funk committed:
MDS-3556: Devops migration to OCP4 (#1755)

* Initial port of templates for ocp4 migration

* Update build configs

* Removing symbolic links

* Update deploy configs

* Add secondary services builds

* Cleanup secondary services dcs

* Add digdag role config

* Tweak psql, nginx and metabase configs

* Tweak digdag, fs, nris and docman configs

* Update workflows for dev deploy

* Remove postgis from ext, already installed with psql10

* Revert "Remove postgis from ext, already installed with psql10"

This reverts commit da8f93a8971b9765d4ea83749a9351498e6075d8.

* Test postgis removal

* Add base psql build for postgis

* Readd postgis ext

* Update psql base dockerfile

* Update yum commands for psql base

* Change psql base epel install

* Update yum flags for psql build

* Update yum urls for psql

* Add sm for repos in psql build

* Test rhsm entitlement in psql build

* Provide explicit package path for psql base

* Update rpm repos in psql build

* Change yum calls for psql build

* Use system resolver for nginx

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Update yum rpm repos

* Disable unneeded yum packages

* Attempt yum rh bugfix

* Change postgis pkg name

* Add additional rpms to psql dockerfile

* Add postgres96 docker image

* Add permission fix for postgres96

* Update fix permission docker path

* Update permissions on postgres96

* Disable postgres explicit entrypoint

* Update postgres96 dockerfile

* Update dockerfile root files for postgres96

* Update ocp configs

* Update volumes for backend

* Increase nginx url hash bucket size

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Remove nginx DNS config

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Add nris etl cronjob

* Update openshift4 dc templates

* Add mds and minespace routes

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Change route to not specify host

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Add env to nginx

* Change default API env variable

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Update frontend dc

* Add minespace configs

* Comment out routes

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Add minespace domain

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Minespace port update

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Minespace service url update

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Update minespace envs

* Use minespace kc client id

* Add docman api url

* Pass through jwt env to backend

* Update github actions for ocp4

* Remove logto file on core-app

Signed-off-by: Paul Dittaro <pdittaro@gmail.com>

* Increase psql pvc

* Passthrough kc resource in deploys

* Increase dc health check timeouts

* Include oracle fdw in postgres96 build

* Adjust postgres deploy probes

* Use devops branch in ocp4 workflows for testing

* Update fs provider url in deployments

* Update build workflow

* Test removal of frontend s2i

* Add s2i image for fs provider

* Temp disable build trigger

* Update fs base image with build entitlements

* Update dotnet dockerfile

* Include rpm in dotnet build

* curl in epel-release in dotnet build

* enable rhel 8 rpms in dotnet build

* Using rhel7 for rpms in dotnet

* Cleanup dotnet base dockerfile

* Add user context to dotnet base image

* Set user in fs provider dockerfile

* weird prefix used on this service.

* Set usergroup permissions in dotnet build

* Cleanup build workflow

* Change user after permission apply in dotnet build

* Enable object store for docman

* Remove imagechange trigger on fe base build

* Update filesystem bc

* This JS syntax wasn't compiling correctly in new env?

* Default docman object store to 0 - adding logic to pipeline

* Update workflow redhat actions with new input names

* Update S3_PREFIX to "/" to match change from 3 weeks ago

* Default docman object store to 1

* Explicit passthrough of docman object store in deploys

* Add test triggers to test/prod workflows

* Update promotion imagestreams list

* Increase replicas to make services HA

* Update tusd s3 prefix

* Update tusd prefix with slash

* Increase resourcing on heavy builds

* Optimize dc resources

* Optimize fs and tusd resources

* Update resources for nris and ms

* set nris replicas to 2 while evaluating quotas

* Update workflows to use develop ref

Co-authored-by: Paul Dittaro <pdittaro@gmail.com>
Co-authored-by: Jason Sy <jasyrotuck@gmail.com>
Co-authored-by: Luke Mitchell <lukegamitchell@gmail.com>

jon-funk committed:
Fix/adjust ocp4 quotas (#1771)

* Adjust ocp4 quotas

* Disable base image builds

jon-funk committed:
Remove configchange triggers from all buildconfigs to prevent double builds (#1772)

jon-funk committed:
Re-add bases build to prevent race condition (#1774)

jon-funk committed:
OCP4 pipeline adjustments (#1775)

* Add success gate to test promotion

* Remove sonarqube from base builds

jon-funk committed:
Adjust tusd url in deployment configs (#1776)

jon-funk committed:
Update nris base path url in deployment configs (#1777)

jon-funk committed:
Correct minespace pipeline params (#1780)

exzizt committed:
MDS-3717: Create Union Rep Party Type + Bug Fixes (#1763)

* Create V2021.06.17.15.09__add_union_rep_party_type.sql

* Add Union Rep + fix bugs/tidy creating/editing of MPAs

* include permittees --> include permit contacts

* Tidy

* Remove debug

* Rename var

* Fix test?

* Fix tests

* Update mine_party_appt_resource.py

* Update test_mpa_permittee_resource.py

* Move checks/change error code

* Update tests

* Fixes?

* Update error codes

* Appt Code needs to be a param to "assign related guid" function because it doesn't exist on the model yet, move validation to save function

* Typo

* Update test_mine_party_appt_resource.py

* Update mine_party_appt.py

* No space

Co-authored-by: Meredith Mumby <meredith.mumby@gmail.com>
Co-authored-by: jon-funk <johnthomasfunk@gmail.com>

jon-funk committed:
Update packages used in ocp4 builds (#1781)

exzizt committed:
Fix bug where NoW activities weren't always returned (#1782)

* Logging with foo

* WIP stuff

* Tidy + remove unused imports

* BUG FIX + tidy

* Remove debug

* Revert

exzizt committed:
Update mine_party_appt_resource.py (#1783)

jon-funk committed:
Correct minespace template param (#1784)

jon-funk committed:
Remove jenkinsfile (#1786)

jon-funk committed:
Add fider app (#1785)

* Add fider configs

* Add fider to workflows

jon-funk committed:
Correct fider route object (#1787)

jon-funk committed:
Dont promote fider because it has a static tag (#1788)

JustinMacaulay committed:
mds 3725 New Document management tab (#1779)

* MDS-3725: added endpoint to update document details

* MDS-3725: WIP UI for Manage documents

* MDS-3275: tests

* MDS-3725: WIP added referral, public comment, consultation documents

* MDS-3725: WIP application documents

* MDS-3725: mapping for application documents

* MDS-3725: final application package

* MDS-3725: fix for navigation, hide part of permit column for final app package

* document changes on the manage documents table

* new doc changes

* styling and functionality for packages

* test updates

* missed snapshot change from dev merge

* updated badge

* updated logic for showing referral/consultation pencils if not in progress

* updated the snapshots

* Descriptions and some small tweaks.

* updated linting

* updated the administrative amendment view to not include submission documents

* updated snapshot tests

* added preamble_date

* missed the dissable buttons in now_documents

* new edit functionality

* PR comments

* added new snapshot tests

* common folder update

Co-authored-by: mhuseinov <mekhti.huseinov.ssi@gmail.com>
Co-authored-by: Meredith Mumby <meredith@freshworks.io>

areyeslo committed:
MDS-3639 Update the HSRC document and MEMP documents for download in Core on the landing page (#1789)

JustinMacaulay committed:
Bug/now documents (#1790)

* bug

* syntax issue

mmumby committed:
vFCBC data bug (#1791)

exzizt committed:
Debugging work and notes (#1793)

Temporary logging to debug an issue

exzizt committed:
Mds 3828 inspectors timeout bug (#1794)

mmumby committed:
ESUPS (#1778)

exzizt committed:
Mds 3828 inspectors timeout bug (#1795)

exzizt committed:
Mds 3828 inspectors timeout bug (#1796)

exzizt committed:
Mds 3828 inspectors timeout bug (#1797)

exzizt committed:
Mds 3828 inspectors timeout bug (#1798)

Fix inspectors caching bug

exzizt committed:
Mds 3828 inspectors timeout bug (Improve query performance) (#1799)

Improve party/inspectors resource query performance

exzizt committed:
Add index to mine_party_appt to improve mine_summary_view performance (#1800)

Add index to mine_party_appt to improve mine_summary_view performance
